test/test_linalg.py

```diff
@@ -54,6 +58,477 @@ def test_det(self, device, dtype):
+            x = torch.randn(25, 25, device=device, dtype=dtype)
+            xn = x.cpu().numpy()
+            for ord in supported_matrix_ords:
+                # TODO: Need to fix abort when nuclear norm is given cdouble input:
+                #       "double free or corruption (!prev) Aborted (core dumped)"
+                if ord == 'nuc' and dtype == torch.cdouble:
+                    continue
+                res = torch.linalg.norm(x, ord, keepdim=keepdim).cpu()
+                expected = np.linalg.norm(xn, ord, keepdims=keepdim)
+                msg = gen_error_message(x.size(), ord, keepdim)
+                self.assertEqual(res.shape, expected.shape, msg=msg)
+                self.assertEqual(res, expected, msg=msg)
+
+        # Test unsupported ords
+        # vector norm
+        x = torch.randn(25, device=device, dtype=dtype)
+        for ord, error_msg in unsupported_vector_ords:
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                torch.linalg.norm(x, ord)
+
+        # matrix norm
+        x = torch.randn(25, 25, device=device, dtype=dtype)
+        for ord, error_msg in unsupported_matrix_ords:
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                torch.linalg.norm(x, ord)
+
+    # Make sure that linalg.norm raises an error if dim is an integer
+    # TODO: When integer dims are supported in norm, remove this test
+    def test_norm_dim_int_error(self, device):
+        input = torch.randn(10, device=device)
+        with self.assertRaisesRegex(TypeError, r'linalg_norm\(\) received an invalid combination of arguments'):
+            torch.linalg.norm(input, dim=0)
+
+    # Test that linal.norm gives the same result as numpy when inputs
+    # contain extreme values (inf, -inf, nan)
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_norm_extreme_values(self, device):
+        vector_ords = [0, 1, 2, 3, inf, -1, -2, -3, -inf]
+        matrix_ords = ['fro', 'nuc', 1, 2, inf, -1, -2, -inf]
+        vectors = []
+        matrices = []
+        for pair in itertools.product([inf, -inf, 0.0, nan, 1.0], repeat=2):
+            vectors.append(list(pair))
+            matrices.append([[pair[0], pair[1]]])
+            matrices.append([[pair[0]], [pair[1]]])
+        for vector in vectors:
+            x = torch.tensor(vector).to(device)
+            x_n = x.cpu().numpy()
+            for ord in vector_ords:
+                msg = f'ord={ord}, vector={vector}'
+                result = torch.linalg.norm(x, ord=ord)
+                result_n = np.linalg.norm(x_n, ord=ord)
+                self.assertEqual(result, result_n, msg=msg)
+
+        # TODO: Need to fix these cases
+        def is_broken_matrix_norm_case(ord, x):
+            if self.device_type == 'cuda':
+                if x.size() == torch.Size([1, 2]):
+                    if ord in ['nuc', 2, -2] and isnan(x[0][0]) and x[0][1] == 1:
```

Comment:
I just found these broken extreme value cases when I added 1.0 to the pair values above. Currently looking for the cause.

Reply:
This is an issue with `torch.svd`:

```
>>> torch.svd(torch.tensor([[float('nan'), 1.0]]))
torch.return_types.svd(
U=tensor([[1.]]),
S=tensor([nan]),
V=tensor([[nan],
        [nan]]))
>>> torch.svd(torch.tensor([[float('nan'), 1.0]]).cuda())
torch.return_types.svd(
U=tensor([[1.]], device='cuda:0'),
S=tensor([1.4142], device='cuda:0'),
V=tensor([[nan],
        [nan]], device='cuda:0'))
```

@mruberry, I could probably detect this failing case after calling `svd` and change the result accordingly. But whether or not we decide to do that, I think `svd` should probably be fixed.

Reply:
Would you file an issue for SVD? Then in the test you can add a comment pointing to the issue as an explanation for why it's avoided.

Reply:
Sure thing, I just filed #43567